packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -2491,4 +2493,43 @@ NumberFormat.decimalPatternDigits(
     setupLocalizations(<String, String>{ 'en': dollarSignWithSelect });
     expect(getGeneratedFileContent(locale: 'en'), contains(r'\$nice_bug\nHello Bug! Manifistation #1 $_temp0'));
   });
+
+  testWithoutContext('can generate method with named argument', () {
+    const String arbFile = '''
+{
+  "helloName": "Hello {name}!",
+  "@helloName": {
+    "description": "A more personal greeting",
+    "placeholders": {
+      "name": {
+        "type": "String",
+        "description": "The name of the person to greet"
+      }
+    }
+  },
+  "helloNameAndAge": "Hello {name}! You are {age} years old.",
+  "@helloNameAndAge": {
+    "description": "A more personal greeting",
+    "placeholders": {
+      "name": {
+        "type": "String",
+        "description": "The name of the person to greet"
+      },
+      "age": {
+        "type": "int",
+        "description": "The age of the person to greet"
+      }
+    }
+  }
+}
+    ''';
+    setupLocalizations(<String, String>{ 'en': arbFile }, useNamedArgument: true);
+    final String localizationsFile = getGeneratedFileContent(locale: 'en');
+    expect(localizationsFile, containsIgnoringWhitespace(r'''
```

Comment:
do we test that this compiles?

Reply:
Ahh, looks like we DO have a test that actually runs a flutter app, how about you exercise this new feature somewhere in this file: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart#L332

Reply:
I have added a test for this.